### PR TITLE
Bug fix, small optimization

### LINKED ISSUE #541: [BUG] Join message for NPCs
**Describe the bug**
When you create NPC with Citizen's API and use EntityType.PLAYER the plugin shows the message 'xyz is visiting your island' even for the NPC. 

**To Reproduce**
1. Create NPC on the island (type=player, give it a name)
2. Go on your island, it'll say 'xyz is visiting your island'

**Console Error**
none

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Rhonim**
- Fix Netherwarts error upon growth (Please check this, I can only assume you want to count nether warts as crops and have them benefit from the islandFarmingBooster
- Optimize getIslands some to have less iterations if it is excluding private islands
- Exclude NPCs  from visiting message (closes #541 )

### CODE REVIEW

**Comment by BomBardyGamer** [comment]
I don't like how you've changed some of the styling here, you've put braces around if statements where it was not necessary and I would rather it have remained like it was before

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 1]
```diff
@@ -110,15 +110,24 @@ public void onMonitorBlockPlace(BlockPlaceEvent event) {
 
     @EventHandler
     public void onBlockClick(PlayerInteractEvent event) {
-        if (event.getAction() != Action.RIGHT_CLICK_BLOCK) return;
+        if (event.getAction() != Action.RIGHT_CLICK_BLOCK) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: There was no need to do this, can you revert this please?

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 2]
```diff
@@ -110,15 +110,24 @@ public void onMonitorBlockPlace(BlockPlaceEvent event) {
...
+            return;
+        }
         final Block block = event.getClickedBlock();
-        if (block == null || block.getType() == Material.AIR) return;
+        if (block == null || block.getType() == Material.AIR) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: Again, no need to separate this, can you undo this please?

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 3]
```diff
@@ -110,15 +110,24 @@ public void onMonitorBlockPlace(BlockPlaceEvent event) {
...
+            return;
+        }
         final Location location = block.getLocation();
         final Island island = IslandManager.getIslandViaLocation(location);
+
         final Material material = block.getType();
         final XMaterial xmaterial = XMaterial.matchXMaterial(material);
-        if (island == null) return;
-        if (island.getPermissions(User.getUser(event.getPlayer())).placeBlocks) return;
+        if (island == null) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: Once again, more changing formatting, can you undo this please?

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 4]
```diff
@@ -132,15 +141,19 @@ public void onBlockClick(PlayerInteractEvent event) {
             if (event.getPlayer().isSneaking() && material == event.getPlayer().getItemInHand().getType() && canStack) {
                 event.setCancelled(true);
                 island.stackedBlocks.compute(location, (loc, original) -> {
-                    if (original == null) return 1 + amount;
+                    if (original == null)
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: Okay, this is just no. Definitely undo this before I merge this.

File: src/main/java/com/iridium/iridiumskyblock/managers/SQLManager.java [hunk 5]
```diff
@@ -10,56 +11,68 @@
...
 public class SQLManager {
-    private final HikariDataSource hikariDataSource = new HikariDataSource();
+    private HikariDataSource hikariDataSource;
 
     public SQLManager() {
         setupConnection();
     }
 
     private void setupConnection() {
+        if (hikariDataSource != null) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/managers/SQLManager.java]: Can this be a single line please? To match the rest of the formatting

File: src/main/java/com/iridium/iridiumskyblock/managers/UserManager.java [hunk 6]
```diff
@@ -21,28 +21,30 @@ public static User getUser(String uuid) {
 
     //Gets a user from UUID
     public static User getUser(UUID uuid) {
-        if (cache.containsKey(uuid)) return cache.get(uuid);
-        try {
-            Connection connection = IridiumSkyblock.getInstance().getSqlManager().getConnection();
-            PreparedStatement statement = connection.prepareStatement("SELECT * FROM users WHERE UUID =?;");
+        if (cache.containsKey(uuid)) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/managers/UserManager.java]: Same with this, single line please, for consistency

File: src/main/java/com/iridium/iridiumskyblock/schematics/SchematicData.java [hunk 7]
```diff
@@ -56,43 +56,43 @@ public SchematicData(File file, short width, short length, short height, List<Ta
...
-                return new SchematicData(file, width, length, height, TileEntities, blockdata, palette, version);
-            } else if (version == 2) {
-                List<Tag> BlockEntities = getChildTag(schematic, "BlockEntities", ListTag.class).getValue();
-                return new SchematicData(file, width, length, height, BlockEntities, blockdata, palette, version);
             } else {
-                return new SchematicData(file, width, length, height, blockdata, palette, version);
-            }
+                List<Tag> TileEntities = getChildTag(schematic, "TileEntities", ListTag.class).getValue();
+                String materials = getChildTag(schematic, "Materials", StringTag.class).getValue();
+                if (!materials.equals("Alpha")) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/schematics/SchematicData.java]: Single line please, for consistency

File: src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java [hunk 8]
```diff
@@ -1,15 +1,59 @@
...
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
 import com.iridium.iridiumskyblock.api.IridiumSkyblockAPI;
 import com.iridium.iridiumskyblock.api.IridiumSkyblockReloadEvent;
 import com.iridium.iridiumskyblock.bank.BankItem;
 import com.iridium.iridiumskyblock.commands.CommandManager;
-import com.iridium.iridiumskyblock.configs.*;
-import com.iridium.iridiumskyblock.gui.*;
-import com.iridium.iridiumskyblock.listeners.*;
+import com.iridium.iridiumskyblock.configs.Bank;
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java]: Any idea why all the imports got unwrapped?

File: src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java [hunk 9]
```diff
@@ -22,7 +66,13 @@
 import com.iridium.iridiumskyblock.schematics.WorldEdit6;
 import com.iridium.iridiumskyblock.schematics.WorldEdit7;
 import com.iridium.iridiumskyblock.serializer.Persist;
-import com.iridium.iridiumskyblock.support.*;
+import com.iridium.iridiumskyblock.support.AdvancedSpawners;
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java]: Unwrapped imports?

File: src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java [hunk 10]
```diff
@@ -37,15 +87,26 @@
 import org.bukkit.plugin.RegisteredServiceProvider;
 import org.bukkit.plugin.java.JavaPlugin;
 
-import java.io.*;
+import java.io.BufferedOutputStream;
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java]: More unwrapped imports

File: src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java [hunk 11]
```diff
@@ -37,15 +87,26 @@
...
+import java.io.OutputStream;
 import java.lang.reflect.Field;
 import java.net.URL;
 import java.net.URLConnection;
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
-import java.util.*;
+import java.util.ArrayList;
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java]: More unwrapped imports

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 12]
```diff
@@ -110,15 +110,24 @@ public void onMonitorBlockPlace(BlockPlaceEvent event) {
...
         final Island island = IslandManager.getIslandViaLocation(location);
+
         final Material material = block.getType();
         final XMaterial xmaterial = XMaterial.matchXMaterial(material);
-        if (island == null) return;
-        if (island.getPermissions(User.getUser(event.getPlayer())).placeBlocks) return;
+        if (island == null) {
+            return;
+        }
+        if (!island.getPermissions(User.getUser(event.getPlayer())).placeBlocks) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: Another if statement that should be a single line

**Review by BomBardyGamer — CHANGES_REQUESTED** [review]
Overall decent PR. The database system is in the process of being switched to an object-relational mapping system, but if you make these changes, I'll merge it. Thanks!

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java [hunk 13]
```diff
@@ -23,9 +23,15 @@ public void onPlayerTeleport(PlayerTeleportEvent event) {
         final Location fromLocation = event.getFrom();
         if (!IslandManager.isIslandWorld(toLocation)) return;
         final Island toIsland = IslandManager.getIslandViaLocation(toLocation);
-        if (toIsland == null) return;
+        if (toIsland == null) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java:26]: Enough of these.

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java [hunk 14]
```diff
@@ -23,9 +23,15 @@ public void onPlayerTeleport(PlayerTeleportEvent event) {
...
+            return;
+        }
 
         final Player player = event.getPlayer();
+        if (player.hasMetadata("NPC")) {
```
Comment (BomBardyGamer) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java:31]: Enough of these.

**Review by BomBardyGamer — CHANGES_REQUESTED** [review]
More if statements on separate lines

**Comment by BomBardyGamer** [comment]
Other than the styling inconsistencies I've addressed, this all looks fine to me

**Comment by BomBardyGamer** [comment]
I know your try-with-resources PR is mostly to blame for all of these, but I would like if you could revert those, as I don't believe @PeachesMLG properly reviewed your original PR

**Comment by Rhonim** [comment]
I'll be frank, I'm not going back and changing that. The brackets are added for readability sake. Braceless style is generally discouraged and not used for various reasons, especially in the cooperate world. Import wise, same thing, there are many reasons I have my IDE default to not using *, if you don't like a long import list then you may want to switch to an IDE that allows you to minimize it.

**Pull request merged**